src/Microsoft.DotNet.Interactive/KernelCommandPipeline.cs

```diff
@@ -37,15 +37,15 @@ private void EnsureMiddlewarePipelineIsInitialized()
         {
             EnsureMiddlewarePipelineIsInitialized();
 
-            command.TryAddToRoutingSlip(_kernel.GetKernelUri());
-            
             try
             {
-                await _pipeline(command, context, (_, _) => Task.CompletedTask);
+                await _pipeline(command, context, (_, __) => Task.CompletedTask);
+                command.TryAddToRoutingSlip(_kernel.GetKernelUri());
```

Comment:
Rather than repeat this line of code, why not do it after the `try`/`catch` block?